static/js/src/advantage/subscribe/react/APICalls/useFreeTrial.tsx

```diff
@@ -0,0 +1,68 @@
+import { useMutation } from "react-query";
+import {
+  postGuestFreeTrial,
+  postLoggedInFreeTrial,
+} from "../../../contracts-api";
+import useProduct from "./useProduct";
+
+import { FormValues } from "../utils/utils";
+
+const FreeTrial = () => {
+  const { product, quantity } = useProduct();
+
+  const mutation = useMutation(async (formData: FormValues) => {
+    if (!product) {
+      throw new Error("Product missing");
+    }
+
+    const {
+      name,
+      organisationName,
+      email,
+      address,
+      city,
+      country,
+      postalCode,
+      usState,
+      caProvince,
+    } = formData;
+
+    const addressObject = {
+      city: city,
+      country: country,
+      line1: address,
+      postal_code: postalCode,
+      state: country === "US" ? usState : caProvince,
+    };
+
+    let res;
+
+    if (window.isGuest) {
```

Comment:
Could you think of a better place/way to handle this? Ideally this should be decoupled from this hook and passed as a parameter. Look at: https://reactjs.org/docs/hooks-custom.html#tip-pass-information-between-hooks

Looks like @huwshimi already has done some work on a hook for window data that you could use:
https://github.com/petermakowski/ubuntu.com/blob/7d99f5bd8749cf1d620151ee85a6a6dd63628440/static/js/src/advantage/react/hooks/useLoadWindowData.ts#L26

Reply:
(Not insisting on doing this as part of this PR)

Reply:
I don't know that it should be decoupled from the hook. The hook handles starting a trial and decides which API endpoint to call. I think it's less error-prone to let the hook handle it as it's part of its logic.

I'm not super convinced that we gain something by adding a hook to forward all the data from the window object into a ReactQuery query. I'm not sure ReactQuery should be used as a general store.

Reply:
Fair points.

What I wanted to highlight is that it would be beneficial to  keep the `isGuest` state in React. I think we should move away from using globals like that as much as possible.